templates/fine-tune-function-calling/README.md

```diff
@@ -0,0 +1,419 @@
+```
+
+
+```python
+# Anyscale Endpoints are OpenAI compatible
+client = openai.OpenAI(
+    base_url = ANYSCALE_API_BASE,
+    api_key = "esecret_yourKeyHere" 
+)
+```
+
+Let's now launch a fine-tuning job for 4 epochs. The expected time for this job is < 3 hours.
+
+
+```python
+# other parameters like context length will be chosen appropriately based on dataset size
+client.fine_tuning.jobs.create(
+    model="meta-llama/Meta-Llama-3-8B-Instruct",
+    hyperparameters={"n_epochs": 4},
+    # replace with the actual file ids!
+    training_file="file_trainingFileId",
+    validation_file="file_validationFileId",
+)
+```
+
+# Step 3: Serving
+
+## Step 3(a): Finetuned on the Anyscale Platform
+
+Make a note of the final checkpoint after fine-tuning (this should be the last line in the logs). You can now spin up the "Deploy LLMs" template which has all the instructions and required dependencies to serve your finetuned model efficiently. You will find the tutorials on [serving LoRA models](https://github.com/anyscale/templates/blob/main/templates/endpoints_v2/examples/lora/DeployLora.ipynb) (if applicable) and on deploying a [custom model](https://github.com/anyscale/templates/blob/main/templates/endpoints_v2/examples/CustomModels.ipynb) helpful. Once you have set up your fine-tuned model as an Anyscale Service, note down the base URL and API key and place them here.
+
+
+```python
+# to be run only if you finetuned on the Anyscale platform
+ANYSCALE_API_KEY="service-api-key-here"
+ANYSCALE_API_BASE="service-url-here" 
+if not ANYSCALE_API_BASE.endswith("/"):
+    ANYSCALE_API_BASE += "/"
+ANYSCALE_API_BASE += "v1"
+# enter the model id here. This would be different depending on whether you performed LoRA or full parameter fine-tuning.
+# Example: meta-llama/Meta-Llama-3-8B-Instruct:mysuffix:myid 
+MODEL_ID = "ModelIdHere"
+```
+
+## Step 3(b): Finetuned through serverless endpoints
+
+To serve the fine-tuned model, you just need to navigate to the "Serving" section on the Anyscale Platform. Your fine-tuned model should already be visible in the list of available models! 
+
+<p align="center">
+  <img src="./assets/serving_endpoints.png" alt="Serve Endpoints">
+</p>
+
+
+As in the above image, click on the three dots and then click on "Query". This will provide you the starter code to interact with the model via curl, python, etc. 
+
+<p align="center">
+  <img src="./assets/serve_api_key.png" alt="API Key">
+</p>
+
+
+
+```python
+## This is only if you finetuned through serverless endpoints
+ANYSCALE_API_BASE = "https://api.endpoints.anyscale.com/v1"
+ANYSCALE_API_KEY = "esecret_yourKeyHere"
+MODEL_ID = "yourModelIdHere" # make sure to not add a stray slash "/"" at the end!
+```
+
+### (Optional) Try out the model via Playground
+
+You can try out your new model in the Playground: https://console.anyscale.com/v2/playground . In the model dropdown, you should be able to see your finetuned model as shown below
```

Comment:
ok playground experience is limited to sdk finetuned models only, so make it clear.

Reply:
Good idea. I just kept it to be a subsection for the serverless endpoints finetuning, but added an explicit note.